### PR TITLE
Investigating a new respiratory virus outbreak in Santa's Workshop

### PR DESCRIPTION
**Pull request by mdperry**
I included three data files (none of them are particularly large), thinking that if anybody wanted to try this out then it would be easier to provide the files than to ask them to search the web for the right databases.  Both of the RSV-A sequence files are available as publicly released entries (each one is in a different database).  If needed I could also provide a link to the GitHub repo that is the source of the bed file, it is also in the public domain.

I reviewed how other authors had laid out and formatted the code and the descriptions/text, and tried to copy/emulate that, but I don't actually know what this would all look like in the final format.

I am open to any feedback/suggestions/comments or other things that could improve any part of this submission.

### CODE REVIEW

**Comment by oalders** [comment]
127 0 0 1_7007_2024-12-01 html-fs8

File: 2024/incoming/investigating-a-new-respiratory-virus-outbreak-in-santa-s-workshop.pod [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+}
+close $FH3;
+
+exit;
+
+=end perl
+
+=head2 Conclusion
+
+When you run the script save the output to file, and then use C<grep undef> to see if there were
```
Comment (oalders) [inline 2024/incoming/investigating-a-new-respiratory-virus-outbreak-in-santa-s-workshop.pod:224]:
```suggestion
When you run the script, save the output to file, and then use C<grep undef> to see if there were
```

File: 2024/incoming/investigating-a-new-respiratory-virus-outbreak-in-santa-s-workshop.pod [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+any coordinates which could NOT be successfully mapped.  If there are then you can use C<grep -v undef>
```
Comment (oalders) [inline 2024/incoming/investigating-a-new-respiratory-virus-outbreak-in-santa-s-workshop.pod:225]:
```suggestion
any coordinates which could NOT be successfully mapped.  If there are any then you can use C<grep -v undef>
```

**Review by oalders — COMMENTED** [review]
I love it! I just made a few suggestions. My only real feedback is that it might be helpful to link to definitions of terms and short forms which people outside of this sphere may not be familiar with. That's not essential to understanding the Perl, but a little more background could make for an easier read for an outsider.

I'm quite happy to see this contribution as I don't think we've traditionally had this specialized kind of case covered in the Advent Calendar. I think it's a great addition to this edition.

**Comment by mdperry** [comment]
Cool!  Seriously, the layout and rendering looks just like I hoped and imagined.  I haven't looked at it for several days, and was also pleased to see that it reads/flows they way I wanted.  There were some small typos, but I think you found and corrected all of them.
Do I have to take some action for all of these comments/suggestions that you made?  Like just click on the 'Resolve conversation' buttons for each one?

I think your point about linking to definitions is a great idea.  Reliance on jargon is always (often) a challenge for technical writing and it can alienate readers.  So I will work on making a list of the terms that may be unfamiliar, and then see if I can find some good definitions to link to.  Another option that is sometimes used in Review Articles is to create a side box (or side boxes) with definitions, but embedding links is probably easier.

**Comment by oalders** [comment]
> Do I have to take some action for all of these comments/suggestions that you made? Like just click on the 'Resolve conversation' buttons for each one?

You could click on "commit suggestion" and that would create an individual commit for each suggestion. Or could do it manually and then click "resolve" on each suggestion.

**Comment by oalders** [comment]
@mdperry is this ready to ship?

**Comment by mdperry** [comment]
Yes, please.  I think so.  The most recent commit should have those links inserted (to explain terms), and I realized that I kept typing 'RSV-A' but officially it should be 'RSVA' so I changed that everywhere.  I noticed just now (above) that this latest commit is not 'Verified', but I am unclear if there is an action I need to take to make it Verified (or if that is required).

**Comment by oalders** [comment]
The verified status doesn't matter here. 😄  Thanks for the update. I'll take another look when I have a moment.

**Comment by oalders** [comment]
Thanks @mdperry \o/

**Pull request merged**